src/coreclr/src/vm/corhost.cpp

```diff
@@ -819,11 +796,25 @@ HRESULT CorHost2::_CreateDelegate(
             }
         }
 
-        if (pMD==NULL || !pMD->IsStatic() || pMD->ContainsGenericVariables())
+        if (pMD==NULL || !pMD->IsStatic() || pMD->HasClassOrMethodInstantiation())
             ThrowHR(COR_E_MISSINGMETHOD);
 
-        UMEntryThunk *pUMEntryThunk = pMD->GetLoaderAllocator()->GetUMEntryThunkCache()->GetUMEntryThunk(pMD);
-        *fnPtr = (INT_PTR)pUMEntryThunk->GetCode();
+        if (pMD->HasNativeCallableAttribute())
+        {
+            if (NDirect::MarshalingRequired(pMD, pMD->GetSig(), pMD->GetModule()))
+                ThrowHR(COR_E_INVALIDPROGRAM);
```

Comment:
This API returns an `HRESULT`, is there a reason we are throwing?

Reply:
No particular reason. I am just following the style of the existing code.